src/backend/vulkan/src/physical_device.rs

```diff
@@ -1395,4 +1412,281 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
         }
         true
     }
+
+    unsafe fn enumerate_available_displays(
+        &self,
+    ) -> Result<Vec<display::Display<Backend>>, display::DisplayError> {
+        let display_extension = match &self.instance.display {
+            Some(display_extension) => display_extension,
+            None => {
+                error!("Direct display feature not supported");
+                return Err(display::DisplayError::UnsupportedFeature);
```

Comment:
Sure. I have modified it accordingly. I have left the `error!` outputs based on the values returned by inner Vulkan functions, so they will help to debug in case of errors. Tell me if you prefer them to be removed too.